tests/sanity/test_cilium_rock.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright 2025 Canonical, Ltd.
+#
+
+import shlex
+
+import pytest
+from k8s_test_harness.util import docker_util, env_util, fips_util
+from test_util import config, rock
+
+IMAGE_NAME = "cilium"
+IMAGE_BASE = f"ghcr.io/canonical/{IMAGE_NAME}"
+IMAGE_ENTRYPOINT = "cilium-agent --version"
+
+
+@pytest.fixture
+def skip_if_nonfips(GOFIPS, image_version):
+    rockcraft_yaml = (
+        (config.REPO_PATH / image_version / "cilium" / "rockcraft.yaml")
+        .read_text()
+        .lower()
+    )
+
+    if GOFIPS == 1 and not fips_util.is_fips_rock(rockcraft_yaml):
+        pytest.skip("Skipping because this version of rock is not built for FIPS")
```

Comment:
I'd suggest to just cache the rockcraft.yaml and handle the skip in the actual test function.
```suggestion
@cache
def get_rockcraft_yaml(GOFIPS, image_version):
    rockcraft_path = config.REPO_PATH / image_version / "cilium" / "rockcraft.yaml"
    return rockcraft_path.read_text().lower()
```

Reply:
Done, thank you